### PR TITLE
Update the registry server we test against from 2.6 to 2.8

### PR DESCRIPTION
**Pull request by mtrmac**
... primarily so that it can support OCI artifacts.

2.8 already seems to exist in the repo.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by mtrmac** [comment]
Those are related — `htpasswd` is no longer included in the `registry` image. So we have to find it elsewhere; it seems that it might be available in the host OS, so let’s try that.

**Comment by mtrmac** [comment]
Full disclosure: I hit “re-run test” a few times on things that looked like timing issues or not related. I’m not sure if that is common in this project or frowned upon.

Notably at least one of the runs reported a flake on “podman push to local registry with authorization” — the registry logged a successful startup but subsequent accesses were refused, but the in-test re-run succeeded. I don’t think that’s a newly introduced bug but it’s possible that the change now makes some pre-existing race more likely (because we now just run a single external command instead of starting a container)

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@vrothberg @giuseppe @flouthoc PTAL

File: test/e2e/login_logout_test.go [hunk 1]
```diff
@@ -80,12 +80,12 @@ var _ = Describe("Podman login and logout", func() {
...
 		setup.WaitWithDefaultTimeout()
 
-		session = podmanTest.Podman([]string{"run", "-d", "-p", strings.Join([]string{strconv.Itoa(port), strconv.Itoa(port)}, ":"),
+		session := podmanTest.Podman([]string{"run", "-d", "-p", strings.Join([]string{strconv.Itoa(port), strconv.Itoa(port)}, ":"),
 			"-e", strings.Join([]string{"REGISTRY_HTTP_ADDR=0.0.0.0", strconv.Itoa(port)}, ":"), "--name", "registry", "-v",
 			strings.Join([]string{authPath, "/auth:Z"}, ":"), "-e", "REGISTRY_AUTH=htpasswd", "-e",
 			"REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm", "-e", "REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd",
 			"-v", strings.Join([]string{certPath, "/certs:Z"}, ":"), "-e", "REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt",
-			"-e", "REGISTRY_HTTP_TLS_KEY=/certs/domain.key", "registry:2.6"})
+			"-e", "REGISTRY_HTTP_TLS_KEY=/certs/domain.key", "registry:2.8"})
```
Comment (Luap99) [inline test/e2e/login_logout_test.go:88]: This should use the REGISTRY_IMAGE variable

File: test/e2e/login_logout_test.go [hunk 2]
```diff
@@ -249,7 +249,7 @@ var _ = Describe("Podman login and logout", func() {
 			strings.Join([]string{authPath, "/auth:z"}, ":"), "-e", "REGISTRY_AUTH=htpasswd", "-e",
 			"REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm", "-e", "REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd",
 			"-v", strings.Join([]string{certPath, "/certs:z"}, ":"), "-e", "REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt",
-			"-e", "REGISTRY_HTTP_TLS_KEY=/certs/domain.key", "registry:2.6"})
+			"-e", "REGISTRY_HTTP_TLS_KEY=/certs/domain.key", "registry:2.8"})
```
Comment (Luap99) [inline test/e2e/login_logout_test.go:252]: same here

Comment (mtrmac) [reply test/e2e/login_logout_test.go]: Thanks, fixed both (and also `hack/podman-registry`).

**Comment by vrothberg** [comment]
Note that this PR is already part of https://github.com/containers/podman/pull/15108, so I am closing this one here to delay our planet's dooms day by a nano second or two.

**Pull request closed without merging**